Documentation/sources/content-card-ui/tutorial/listening-content-card-events.md

```diff
@@ -0,0 +1,108 @@
+
+```kotlin
+@Composable
+private fun AepContentCardList(viewModel: AepContentCardViewModel) {
+  // Create the ContentCardUIEventListener
+  val contentCardCallback = ContentCardCallback()
+  // Collect the state from ViewModel
+  val aepUiList by viewModel.aepUIList.collectAsStateWithLifecycle()
+  
+  // Create row with composables from AepUI instances
+  LazyRow {
+    items(aepUiList) { aepUI ->
+                               when (aepUI) {
+                                 is SmallImageUI -> {
+                                   val state = aepUI.getState()
+                                   if (!state.dismissed) {
+                                     SmallImageCard(
+                                       ui = aepUI,
+                                       style = smallImageCardStyleRow,
+                                       // provide the ContentCardUIEventListener as a parameter to the 				  																	 // ContentCardEventObserver
+                                       observer = ContentCardEventObserver(contentCardCallback)
+                                     )
+                                   }
+                                 }
+                               }
+                              }
+  }
+}    
+    
+class ContentCardCallback: ContentCardUIEventListener {
+  override fun onDisplay(aepUI: AepUI<*, *>) {
+    Log.d("ContentCardCallback", "onDisplay")
+  }
+
+  override fun onDismiss(aepUI: AepUI<*, *>) {
+    Log.d("ContentCardCallback", "onDismiss")
+  }
+
+  override fun onInteract(
+    aepUI: AepUI<*, *>,
+    interactionId: String?,
+    actionUrl: String?
+  ): Boolean {
+    Log.d("ContentCardCallback", "onInteract $interactionId $actionUrl")
+    // If the url is handled here, return true
+    return false
+  }
+}
+```
+
+### Handling actionable URLs
+
+The `onInteract` method provides an optional `actionURL` parameter associated with the interaction event. The return value of this method determines how the URL is handled.
+
+- Return `true` if your application has successfully handled the URL. This indicates to the SDK that no further action is needed.
+
+- Return `false` to allow the SDK to process the URL.
+
+```kotlin
+override fun onInteract(
+  aepUI: AepUI<*, *>,
+  interactionId: String?,
+  actionUrl: String?
+): Boolean {
+  actionUrl?.let { 
+    // handle action url here
+    return true
+  }
+  return false
+}
+```
+
+### Removing content cards on dismiss
+
+> *__IMPORTANT__* - Removing a dismissed content card from the UI is the responsibility of the app developer.
+
+To remove a content card from the UI when the user dismisses it, implement the onDismiss method:
+
+```kotlin
+lateinit var aepUiList: State<List<AepUI<*, *>>>   
+override fun onDismiss(aepUI: AepUI<*, *>) {
+  aepUiList = viewModel.aepUIList.collectAsStateWithLifecycle()
+  val displayedTemplate = aepUI.getTemplate() as SmallImageTemplate
+  aepUiList.value.filterNot { it.getTemplate() == displayedTemplate }
+}
+```
```

Comment:
Do we need this? We will be providing the `AepList` / `ContentCardList` very soon in the beta, isn't it?

Reply:
you are right, i will remove this.